### PR TITLE
Mkulakow/full enable grpc rest mp

### PR DESCRIPTION
**Pull request by michalkulakowski**

### CODE REVIEW

File: src/mediapipe_internal/mediapipegraphexecutor.cpp [hunk 1]
```diff
@@ -211,11 +211,51 @@ static const KFSDataType& MPPrecisionToKFSPrecision(::mediapipe::Tensor::Element
         SPDLOG_DEBUG(details);                                                                              \
         return Status(StatusCode::UNKNOWN_ERROR, std::move(details));                                       \
     }
+
+#define COPY_INPUT_VALUE_BY_VALUE(TYPE, PROTO_PREFIX)                                                                                       \
+    TYPE* ptr = reinterpret_cast<TYPE*>(data);                                                                                              \
+    /* TODO check if has content correct size*/                                                                                             \
```
Comment (atobiszei) [inline src/mediapipe_internal/mediapipegraphexecutor.cpp:217]: TODO to remove?

File: src/mediapipe_internal/mediapipegraphexecutor.cpp [hunk 2]
```diff
@@ -211,11 +211,51 @@ static const KFSDataType& MPPrecisionToKFSPrecision(::mediapipe::Tensor::Element
...
+    const auto& input = request.inputs(inputIndex);                                                                                         \
+    if (!input.has_contents()) {                                                                                                            \
+        return Status(StatusCode::INVALID_CONTENT_SIZE, "Input does not have input tensor contents field");                                 \
+    }                                                                                                                                       \
+    const auto& contents = input.contents();                                                                                                \
+    if (!contents.PROTO_PREFIX##_contents_size()) {                                                                                         \
```
Comment (atobiszei) [inline src/mediapipe_internal/mediapipegraphexecutor.cpp:223]: We should validate contents size against buffer that we will be writing into.

File: src/mediapipe_internal/mediapipegraphexecutor.cpp [hunk 3]
```diff
@@ -390,7 +545,11 @@ static Status deserializeTensor(const std::string& requestedName, const KFSReque
     auto requestInputItr = request.inputs().begin();
     OVMS_RETURN_ON_FAIL(getRequestInput(requestInputItr, requestedName, request));
     auto inputIndex = requestInputItr - request.inputs().begin();
-    auto& bufferLocation = request.raw_input_contents().at(inputIndex);
+    if(request.raw_input_contents_size() <= inputIndex){
+        SPDLOG_DEBUG("Data should be located in raw_input_contents if node input tag is IMAGE");
```
Comment (atobiszei) [inline src/mediapipe_internal/mediapipegraphexecutor.cpp:549]:
```suggestion
        SPDLOG_DEBUG("Data should be located in raw_input_contents if graph input tag is IMAGE");
```

File: src/mediapipe_internal/mediapipegraphexecutor.cpp [hunk 4]
```diff
@@ -390,7 +545,11 @@ static Status deserializeTensor(const std::string& requestedName, const KFSReque
...
+        return StatusCode::MEDIAPIPE_EXECUTION_ERROR;
+    }
+    auto& bufferLocation = request.raw_input_contents().at(inputIndex);  // TODO probably doesn't make sense to handle input.content()
```
Comment (atobiszei) [inline src/mediapipe_internal/mediapipegraphexecutor.cpp:552]: TODO to remove

File: src/mediapipe_internal/mediapipegraphexecutor.cpp [hunk 5]
```diff
@@ -425,7 +584,7 @@ static Status deserializeTensor(const std::string& requestedName, const KFSReque
     }
     size_t elementSize = KFSDataTypeSize(requestInputItr->datatype());
     size_t expectedSize = numberOfChannels * numberOfCols * numberOfRows * elementSize;
-    if (bufferLocation.size() != expectedSize) {
+    if (bufferLocation.size() != expectedSize) {  // TODO FIXME probably doesn't make sense to support since image should be continuous
```
Comment (atobiszei) [inline src/mediapipe_internal/mediapipegraphexecutor.cpp:587]: TODO to remove?

File: src/test/mediapipe/graphpassthrough.pbtxt [hunk 6]
```diff
@@ -0,0 +1,23 @@
+#
+# Copyright 2023 Intel Corporation
```
Comment (atobiszei) [inline src/test/mediapipe/graphpassthrough.pbtxt:2]:
```suggestion
# Copyright 2024 Intel Corporation
```

File: src/tfs_frontend/tfs_utils.cpp [hunk 7]
```diff
@@ -37,6 +37,7 @@ TFSDataType getPrecisionAsDataType(Precision precision) {
         {Precision::I32, TFSDataType::DT_INT32},
         {Precision::I16, TFSDataType::DT_INT16},
         {Precision::I8, TFSDataType::DT_INT8},
+        {Precision::U32, TFSDataType::DT_UINT32},
```
Comment (atobiszei) [inline src/tfs_frontend/tfs_utils.cpp:40]: Check if this is used in `*(UN)SUPPORTED*` precisions list in test.

File: src/test/test_utils.cpp [hunk 8]
```diff
@@ -664,6 +664,10 @@ void randomizePort(std::string& port) {
         *digitToRandomize = '0' + dist(eng);
     }
 }
+std::string getRandomizedPort(std::string port) {
```
Comment (atobiszei) [inline src/test/test_utils.cpp:667]: Why we need this function?

File: src/test/test_utils.hpp [hunk 9]
```diff
@@ -380,6 +380,44 @@ void prepareKFSInferInputTensor(::KFSRequest& request, const std::string& name,
         }
     }
 }
+
+template <>
+inline void prepareKFSInferInputTensor<bool>(::KFSRequest& request, const std::string& name, const std::tuple<ovms::signed_shape_t, const std::string>& inputInfo,
+    const std::vector<bool>& data, bool putBufferInInputTensorContent) {
```
Comment (atobiszei) [inline src/test/test_utils.hpp:386]: This puts bool values in input content regardless of `putBufferInInputTensorContent`. Is it intended and if so why?

Comment (atobiszei) [reply src/test/test_utils.hpp:386]: Throw in case this is false?

File: src/test/mediapipeflow_test.cpp [hunk 10]
```diff
@@ -2822,7 +2830,7 @@ TEST_F(MediapipeFlowStartTest, AsSoonAsMediaPipeGraphDefinitionReadyInferShouldP
     auto start = std::chrono::high_resolution_clock::now();
     while (!isMpReady(servableName) &&
            (std::chrono::duration_cast<std::chrono::seconds>(std::chrono::high_resolution_clock::now() - start).count() < SERVER_START_FROM_CONFIG_TIMEOUT_SECONDS)) {
-        std::this_thread::sleep_for(std::chrono::microseconds(100));
+        std::this_thread::sleep_for(std::chrono::microseconds(600));
```
Comment (atobiszei) [inline src/test/mediapipeflow_test.cpp:2833]: Please check if this sleep is necessary and if it is necessary to have it this long.

File: src/test/mediapipeflow_test.cpp [hunk 11]
```diff
@@ -2839,6 +2847,453 @@ TEST_F(MediapipeFlowStartTest, AsSoonAsMediaPipeGraphDefinitionReadyInferShouldP
...
+    {typeid(int64_t), ovms::Precision::I64},
+    {typeid(int32_t), ovms::Precision::I32},
+    {typeid(int16_t), ovms::Precision::I16},
+    {typeid(int8_t), ovms::Precision::I8},
+    {typeid(bool), ovms::Precision::BOOL},
+    {typeid(double), ovms::Precision::FP64},
+    {typeid(void), ovms::Precision::BIN}};
+
+template <typename T>
+std::vector<T> prepareData(size_t elemCount, T value, bool maxValue) {
```
Comment (mzegla) [inline src/test/mediapipeflow_test.cpp:2865]: Is it possible to set max as default for `value` ? It could simplify the function.

**Review by atobiszei — APPROVED** [review]

**Review by atobiszei — APPROVED** [review]

File: src/test/mediapipeflow_test.cpp [hunk 12]
```diff
@@ -2839,6 +2847,453 @@ TEST_F(MediapipeFlowStartTest, AsSoonAsMediaPipeGraphDefinitionReadyInferShouldP
...
+    {typeid(int64_t), ovms::Precision::I64},
+    {typeid(int32_t), ovms::Precision::I32},
+    {typeid(int16_t), ovms::Precision::I16},
+    {typeid(int8_t), ovms::Precision::I8},
+    {typeid(bool), ovms::Precision::BOOL},
+    {typeid(double), ovms::Precision::FP64},
+    {typeid(void), ovms::Precision::BIN}};
+
+template <typename T>
+std::vector<T> prepareData(size_t elemCount, T value, bool maxValue = true) {
```
Comment (mzegla) [inline src/test/mediapipeflow_test.cpp:2865]: I was thinking about something like:
`(size_t elemCount, T value = std::numeric_limits<T>::max())`

Then by default it sets maxValue, and if you specify the value it sets the value

**Review by mzegla — APPROVED** [review]

**Pull request merged**